### PR TITLE
Rename ConsensusParams to ChainParams

### LINKED ISSUE #513: Rename `ConsensusParams` to `ChainParams`
`ChainParams` + `ValidatorSet` will be _Consensus Parameters_ (Not an explicit Rust struct, but just a concept)

### PR DESCRIPTION
**Pull request by majecty**
Fix #513

### CODE REVIEW

**Comment by junha1** [comment]
As mentioned https://github.com/CodeChain-io/foundry/issues/513#issuecomment-697186767 ,
it is natural to rename `InitChain`/`UpdateChain` to `InitConsensus`/`UpdateConsensus` too.

**Comment by majecty** [comment]
@junha1 I renamed UpdateChain and InitChain.

File: state/src/impls/top_level.rs [hunk 1]
```diff
@@ -356,7 +356,7 @@ impl TopState for TopLevelState {
         self.top_cache.remove_action_data(key)
     }
 
-    fn update_chain_params(&mut self, chain_params: ChainParams) -> StateResult<()> {
+    fn update_consensus_params(&mut self, chain_params: ChainParams) -> StateResult<()> {
```
Comment (junha1) [inline state/src/impls/top_level.rs:359]: It seems you reverted https://github.com/CodeChain-io/foundry/pull/535/commits/7e7013bffa488c4c4f64c6612404eff2d5755a1a#diff-1c89fe2d0239918fff96d25555b8531fR359 even though it takes `ChainParams`. Why?

File: state/src/traits.rs [hunk 2]
```diff
@@ -57,7 +57,7 @@ pub trait TopState {
     fn update_action_data(&mut self, key: &H256, data: Bytes) -> StateResult<()>;
     fn remove_action_data(&mut self, key: &H256);
 
-    fn update_chain_params(&mut self, chain_params: ChainParams) -> StateResult<()>;
+    fn update_consensus_params(&mut self, chain_params: ChainParams) -> StateResult<()>;
```
Comment (junha1) [inline state/src/traits.rs:60]: ditto

File: state/src/impls/top_level.rs (hunk 1 above)
Comment (majecty) [reply state/src/impls/top_level.rs:359]: There was a mistake while fixing commits. I'll fix it.

File: state/src/traits.rs (hunk 2 above)
Comment (majecty) [reply state/src/traits.rs:60]: There was a mistake while fixing commits. I'll fix it.

**Comment by majecty** [comment]
@junha1 I fixed them.

**Review by junha1 — APPROVED** [review]

**Comment by junha1** [comment]
@dynaxis Please note this change

**Pull request merged**